esphome/components/cse7766/cse7766.cpp

```diff
@@ -132,8 +132,24 @@ void CSE7766Component::parse_data_() {
     }
   }
 
+  float energy = 0.0;
+  if (this->energy_sensor_ != nullptr) {
+    if (this->cf_pulses_last_ == 0 && !this->energy_sensor_->has_state()) {
+      this->cf_pulses_last_ = cf_pulses;
+    }
+    uint16_t cf_diff = cf_pulses - this->cf_pulses_last_;
+    this->cf_pulses_total_ += cf_diff;
+    this->cf_pulses_last_ = cf_pulses;
+    // Calculate CF pulses into total energy only if we have Power coefficient to multiply by
+    if (have_power) {
+      energy = this->cf_pulses_total_ * float(power_coeff) / 1000000.0f / 3600.0f;
+      this->energy_sensor_->publish_state(energy);
+    } else if (!this->energy_sensor_->has_state()) {
+      this->energy_sensor_->publish_state(0);
+    }
```

Comment:
Since it appears the power_coeff and cf_pulses values are valid every time (even when power is 0), it would simplify things to publish every time. A filter will be needed to throttle energy regardless.

```suggestion
    energy = this->cf_pulses_total_ * float(power_coeff) / 1000000.0f / 3600.0f;
    this->energy_sensor_->publish_state(energy);
```

Reply:
Yes you are right, I did not check `power_coeff` for `!have_power`. I will check this on my devices first.